mlir/test/Dialect/SCF/transform-ops.mlir

```diff
@@ -270,3 +270,28 @@ module attributes {transform.with_named_sequence} {
     transform.yield
   }
 }
+
+// -----
+
+func.func @normalize_i32_loop() {
+  // CHECK: scf.for %arg0 = %c0_i32 to {{.*}} step %c1_i32  : i32
```

Comment:
Please, properly match constants instead of hardcoding SSA names as those names are not guaranteed to be stable. Also, properly match upper bound, second loop and induction vars.

Reply:
Done for lower bound, step and induction var. As for upper bound, it is hard to match, because it is calculated using a bunch of expressions that involve arith.ceildiv, arith.muli, because of coalescing algorithm (maybe do you have any suggestions on how to match it in a more proper way?). Also, there is no 2nd loop in the output, because two loops were coalesced.
Unfortunately, the normalization function is not exposed, so I use coalescing to check that normalization of the loop works. Without the patch I get:
```
error: unexpected error: 'scf.for' op failed to verify that all of {lowerBound, upperBound, step} have same type
```
on this test case